packages/core/docs/integrate/integration-guide.md

```diff
@@ -5,319 +5,234 @@ The application does not reload automatically after saving the changes in Server
 
-Inside of the composables packages you have to create another directory, next to `src` called `nuxt`. In that directory, we need to place our Nuxt module. The Nuxt module is taking care of anything you want during the integration to be launched: adding plugins, injecting into the build process, creating some aliases, and more. The basic implementation of that module will add just a plugin that will configure our application (using a wrapper that you have already exposed)
+`onCreate` must return an object with at least `config` and `client` properties but it can have any number of custom properties if needed. This object is later available in API endpoints.
 
-Example of plugin
-```js
-// composables/nuxt/plugin.js
-import { integrationPlugin } from '@vue-storefront/core'
-
-const moduleOptions = <%= serialize(options) %>;
+Let's update the `onCreate` method to create and return a new Axios instance.
 
-export default integrationPlugin(({ app, integration }) => {
-  const settings = { api: '/graphql', user: 'root' }
-  integration.configure({ ...moduleOptions, ...settings })
-});
-```
+```typescript
+// packages/api-client/index.server.ts
+import axios from 'axios';
 
-The module:
-```js
-// composables/nuxt/index.js
-export default function (moduleOptions) {
-  this.addPlugin({
-    src: path.resolve(__dirname, './plugin.js'),
-    options: moduleOptions
+const onCreate = (settings) => {
+  const client = axios.create({
+    baseURL: settings.api.url
   });
-}
+
+  return {
+    config: settings,
+    client
+  };
+};
 ```
 
-### Writing factory params
+In the example above we passed `settings.api.url` to `axios.create`, but it's not defined in `middleware.config.js`. Let's add it:
 
-Once you have done the essential configuration of the Nuxt module and plugin, you can proceed to write composable. A good thing to know is you don't have to build it from scratch, we do most of the job for you! Instead, you have to fill in the required functions and properties called `factoryParams`.
+```javascript
+// packages/theme/middleware.config.js
+module.exports = {
+  integrations: {
+    sloth: { // name of your integration
+      location: '@sloth/api/server', // name of your api-client package followed by `/server`
+      configuration: {
+        api: {
+          url: '' // URL of your eCommerce platform
+        }
+      }
+    }
+  }
+};
+```
 
-The `factoryParams` are including functions that you have to implement to provide the functionality of implementing composable, such a fetching something from the API, or triggering other actions.
+## Implement `useProduct` functionality
 
+It's impossible to write a tutorial explaining how to implement every composable because some might differ wildly between the platforms. For this reason, we will explain how to implement `useProduct` composable, `getProduct` API endpoint, and `productGetters` and leave the rest to you.
```

Comment:
```suggestion
It would be impossible to write a tutorial explaining how to implement every composable because some might differ wildly between the platforms. For this reason, we will explain how to implement `useProduct` composable, `getProduct` API endpoint, and `productGetters` to give as much overview as we can.
```
I would avoid saying something like "It's too hard, so do it yourself" :) At least that was my impression/interpretation.